pmacApp/src/pmacCSController.cpp

```diff
@@ -530,22 +532,24 @@ asynStatus pmacCSController::pmacSetAxisScale(int axis, int scale) {
     return asynError;
   }
 
-  this->lock();
-  pA = getAxis(axis);
-  if (pA) {
-    pA->scale_ = scale;
-    setIntegerParam(axis, PMAC_CS_MotorScale_, scale);
-    asynPrint(this->pasynUserSelf, ASYN_TRACE_FLOW,
-              "%s. Setting scale factor of %d on axis %d, on controller %s.\n",
-              functionName, pA->scale_, pA->axisNo_, portName);
-
-  } else {
-    asynPrint(this->pasynUserSelf, ASYN_TRACE_ERROR,
-              "%s: Error: axis %d has not been configured using pmacCreateAxis.\n", functionName,
-              axis);
-    return asynError;
+  if(initialised()) {
+    this->lock();
+    pA = getAxis(axis);
+    if (pA) {
+      pA->scale_ = scale;
+      setIntegerParam(axis, PMAC_CS_MotorScale_, scale);
+      asynPrint(this->pasynUserSelf, ASYN_TRACE_FLOW,
+                "%s. Setting scale factor of %d on axis %d, on controller %s.\n",
+                functionName, pA->scale_, pA->axisNo_, portName);
+
+    } else {
+      asynPrint(this->pasynUserSelf, ASYN_TRACE_ERROR,
+                "%s: Error: axis %d has not been configured using pmacCreateAxis.\n", functionName,
+                axis);
+      return asynError;
```

Comment:
This isn't necessarily a change in this PR (only the `if(initialised())` seems new), but I notice a lock is taken and that there is a `return asynError` **before** unlocking it again... Is that OK or do we risk keeping a mutex locked in an error condition?

Reply:
well spotted.